pkg/log/zap/zap_test.go

```diff
@@ -296,3 +300,128 @@ var _ = Describe("Zap logger setup", func() {
 		})
 	})
 })
+
+var _ = Describe("Zap flag options setup", func() {
+	fromFlags := Options{}
+
+	BeforeEach(func() {
+		flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError) //flags are now reset
```

Comment:
Nit: Rather than using `flag.CommandLine`, would it work to assign `flag.NewFlagSet()` to a separately declared `flag.FlagSet`, and then use the `.Parse()` method on that struct? If so, that seems like it could simplify things.

```go
var _ = Describe("Zap flag options setup", func() {
  var (
    fromFlags Options
    fs flag.FlagSet
  )

  BeforeEach(func() {
    fromFlags = Options{}
    fs = flag.NewFlagSet(...)
  })

  Context("with -zap-devel options provided", func() {
    It("Should set dev=true with relative options.", func() {
      args := []string{"-zap-devel=true"}
      fromFlags.BindFlags(fs)
      fs.Parse(args)
      ...
    })
  })
```

If you go this route, note that the arguments passed to the `.Parse()` method on the FlagSet should not contain the command name.